gpustat/__main__.py

```diff
@@ -11,12 +11,16 @@
 from .core import GPUStatCollection
 
 
-def print_gpustat(json=False, debug=False, **kwargs):
+def print_gpustat(json=False, debug=False, gpu_stat=None, **kwargs):
```

Comment:
It makes less sense to take `gpu_stat` as an argument for this function. I was thinking of a style like (i) we first create a GPUStatCollection instance here and (ii) then print outputs using it. As the same instance should be reused, we can consider running a wait loop inside this function rather than calling it multiple times.

Reply:
What about making this function a generator, so we can keep the `GPUStatCollection` object in the closure.